ui/pages/confirmations/components/confirm/info/shared/gas-fees-section/gas-fees-details.tsx

```diff
@@ -0,0 +1,87 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { Dispatch, SetStateAction } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../../../selectors';
+import { useEIP1559TxFees } from '../../hooks/useEIP1559TxFees';
+import { useSupportsEIP1559 } from '../../hooks/useSupportsEIP1559';
+import { EditGasFeesRow } from './edit-gas-fees-row';
+import { GasFeesRow } from './gas-fees-row';
+import { GasTimings } from './gas-timings';
+import { useFeeCalculations } from './useFeeCalculations';
+
+export const GasFeesDetails = ({
+  setShowCustomizeGasPopover,
+  showAdvancedDetails,
+}: {
+  setShowCustomizeGasPopover: Dispatch<SetStateAction<boolean>>;
+  showAdvancedDetails: boolean;
+}) => {
+  const t = useI18nContext();
+
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { maxFeePerGas, maxPriorityFeePerGas } =
+    useEIP1559TxFees(transactionMeta);
+  const { supportsEIP1559 } = useSupportsEIP1559(transactionMeta);
+
+  const hasLayer1GasFee = Boolean(transactionMeta?.layer1GasFee);
+
+  const {
+    estimatedFiatFee,
+    estimatedNativeFee,
+    l1FiatFee,
+    l1NativeFee,
+    l2FiatFee,
+    l2NativeFee,
+    maxFiatFee,
+    maxNativeFee,
+  } = useFeeCalculations(transactionMeta);
```

Comment:
It will be nicer to break down this hook if possible, it is returning too many fields.

Reply:
The `maxFees` could be separated into another hook but I think this would also duplicate a fair bit of boilerplate, I'm not sure it's worth it in this case.